workbench-core/datasets-ui/src/pages/index.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18nStrings, getFilterCounterText, TableEmptyDisplay, TableNoMatchDisplay, useNotifications, BaseLayout } from '@aws/workbench-core-swb-common-ui';
+import { useCollection } from '@cloudscape-design/collection-hooks';
+import {
+  Box,
+  BreadcrumbGroupProps,
+  Header,
+  PropertyFilter,
+  SpaceBetween,
+  Table
+} from '@cloudscape-design/components';
+import { FlashbarProps } from '@cloudscape-design/components/flashbar';
+import type { NextPage } from 'next';
+import { useRouter } from 'next/router';
+import React, { useState } from 'react';
+import { useDatasets } from '../api/datasets';
+import { columnDefinitions, searchableColumns } from '../datasets-table-config/datasetsColumnDefinitions';
+import { filteringOptions } from '../datasets-table-config/datasetsFilteringOptions';
+import { filteringProperties } from '../datasets-table-config/datasetsFilteringProperties';
+
+export const DatasetsPage: NextPage = () => {
+  // For functions to return content specific to the table
+  const itemType: string = 'dataset';
+  // App settings constant
+
+  const { datasets, areDatasetsLoading } = useDatasets();
+  const router = useRouter();
+  const { message, notificationType } = router.query;
+  const { displayNotification, closeNotification } = useNotifications();
+
+  const [hasInitialNotificationBeenShown, setHasInitialNotificationBeenShown] = useState(false);
+  if (!!message && !!notificationType && !hasInitialNotificationBeenShown) {
+    const datasetMessageId = 'DatasetMessage';
+    const notification = {
+      type: notificationType as FlashbarProps.Type,
+      dismissible: true,
+      dismissLabel: 'Dismiss message',
+      onDismiss: () => {
+        closeNotification(datasetMessageId);
+      },
+      content: message,
+      id: datasetMessageId
+    };
+    displayNotification(datasetMessageId, notification);
+    setHasInitialNotificationBeenShown(true);
+  }
+
+  // App layout constants
+  const breadcrumbs: BreadcrumbGroupProps.Item[] = [
+    {
+      text: 'Service Workbench',
+      href: '/'
+    },
+    {
+      text: 'Datasets',
+      href: '/datasets'
+    }
+  ];
+
+  // Property and date filter collections
+  const { items, filteredItemsCount, collectionProps, propertyFilterProps } = useCollection(datasets, {
+    filtering: {
+      empty: TableEmptyDisplay(itemType),
+      noMatch: TableNoMatchDisplay(itemType),
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
can these `any`'s be typed?

Reply:
Yes - for the `filteringFunction` return type (fixed)
No - for the `UseCollectionResult.items`

Reply:
It looks like when the `any` is removed from the `items` param it infers `DatasetItem` type. Is that not correct?

Reply:
Not sure if you saw my above comment or not